src/System.Private.CoreLib/src/System/Environment.cs

```diff
@@ -155,6 +155,17 @@ public static string ExpandEnvironmentVariables(string name)
         }
 #endif // FEATURE_WIN32_REGISTRY
 
+        [DllImport(JitHelpers.QCall, CharSet = CharSet.Unicode)]
+        private static extern int GetProcessId();
```

Comment:
I did check and on Linux Win32Native.GetCurrentProcessId() doe exist.    What happens is on linux Interop.Libraries.Kernel32 is redirected to "libcoreclr", and these methods are implemented in libcoreclr.so. 

Here is a list of methods that are exported in that DLL (nm -D libcoreclr.so)
```

00000000004e2c10 T CloseHandle
00000000000ccdf0 T coreclr_create_delegate
00000000000ccf00 T coreclr_execute_assembly
00000000000cc8f0 T coreclr_initialize
00000000000cccf0 T coreclr_shutdown
00000000000ccd70 T coreclr_shutdown_2
00000000000cc180 T CoreDllMain
0000000000524fa0 T CoTaskMemAlloc
0000000000524fc0 T CoTaskMemFree
0000000000524fb0 T CoTaskMemRealloc
00000000004ffb20 T CreateEventExW
00000000004ffa90 T CreateEventW
00000000005008b0 T CreateMutexExW
00000000005007b0 T CreateMutexW
00000000005001c0 T CreateSemaphoreExW
0000000000500250 T CreateSemaphoreW
00000000000cc280 T DllMain
00000000004e2960 T DuplicateHandle
00000000004ee9c0 T FormatMessageW
00000000004ee250 T FreeEnvironmentStringsW
00000000004e5c80 T GetACP
00000000000f4210 T GetCLRRuntimeHost
0000000000509f80 T GetCurrentProcessId
000000000050e780 T GetCurrentThreadId
00000000004edfe0 T GetEnvironmentStringsW
00000000004edbf0 T GetEnvironmentVariableW
00000000000cc3e0 T GetMetaDataInternalInterface
00000000000cc3f0 T GetMetaDataInternalInterfaceFromPublic
00000000000cc400 T GetMetaDataPublicInterfaceFromInternal
00000000004e4640 T GetProcAddress
00000000004dccd0 T GetStdHandle
00000000004f0190 T GetSystemInfo
00000000004f01f0 T GlobalMemoryStatusEx
00000000004ebff0 T LocalAlloc
00000000004ec080 T LocalFree
00000000004ec040 T LocalReAlloc
00000000004ce000 T lstrlenA
00000000004ce020 T lstrlenW
00000000000cc2e0 T MetaDataGetDispenser
00000000004e5e90 T MultiByteToWideChar
00000000004ffe70 T OpenEventW
0000000000500d80 T OpenMutexW
0000000000500470 T OpenSemaphoreW
00000000004d4dc0 T OutputDebugStringW
00000000004e5230 T PAL_RegisterModule
00000000004e53f0 T PAL_UnregisterModule
00000000004f06b0 T QueryPerformanceCounter
00000000004f0710 T QueryPerformanceFrequency
00000000004d5da0 T RaiseException
00000000005008e0 T ReleaseMutex
00000000005002e0 T ReleaseSemaphore
00000000004ffd80 T ResetEvent
00000000004edd70 T SetEnvironmentVariableW
00000000004ffbc0 T SetEvent
00000000005243c0 T SysAllocStringLen
0000000000524570 T SysFreeString
00000000005245b0 T SysStringLen
00000000004ea840 T VirtualAlloc
00000000004eb3f0 T VirtualFree
00000000004eb9d0 T VirtualQuery
00000000004e5f20 T WideCharToMultiByte
00000000004dc9f0 T WriteFile
```
Notice that GetCurrentProcessId is one of them.  

Thus I don't believe you need create the QCALL for it, just use Win32Native.GetCurrentProcessId.

Reply:
I have switched to this and will confirm that it works on Linux.

Reply:
I have confirmed that you are correct, and this does work on Linux.